DatabaseAccesser

turtle11235 opened the pull request:
Database access is managed through a middleman access class

turtle11235 committed:
wip database access

turtle11235 committed:
wip database

turtle11235 committed:
merged main

turtle11235 committed:
wip

turtle11235 committed:
wip

turtle11235 committed:
created DatabaseAccessor that is used in routes and sockets

turtle11235 committed:
merged main

turtle11235 committed:
removed out of place reference to User

ZOrangeBandit commented on server/database/User.js:
```diff
@@ -0,0 +1,105 @@
+
+    constructor(userIn){
+        this.validateUser(userIn)
+        if (userIn.isTeacher ){
```
Why is this if statement here

turtle11235 replied:
I actually meant to delete that whole folder

ZOrangeBandit commented on server/routes/users.js:
```diff
@@ -2,67 +2,101 @@ const { v4: uuidv4 } = require('uuid');
-        email: req.body.email,
-        password: req.body.password,
-        displayName: req.body.displayName,
+    var user = {
```
A request's json is stored at req.body, not req. i.e. (req.email -> req.body.email)

ZOrangeBandit commented on server/routes/users.js:
```diff
@@ -2,67 +2,101 @@ const { v4: uuidv4 } = require('uuid');
+            token: ''
+        }
+    }
+    db.createTeacher(user)
```
This usage of the createTeacher method is not the useage defined in the DatabaseAccessor class

ZOrangeBandit commented on server/database/User.js:
```diff
@@ -0,0 +1,105 @@
+            })
+    }
+
+    static
```
This is unneccissary and should be removed

ZOrangeBandit commented on server/database/User.js:
```diff
@@ -0,0 +1,105 @@
+            user.displayName = user.name;
+        }
+
+        // if 
```
remove this comment

ZOrangeBandit reviewed (changes requested):
There are a couple issues and peculiar changes that I think should be addressed before merging

ZOrangeBandit committed:
Fixed the issues that I saw in the pull request

ZOrangeBandit committed:
Moved updates from database.js models to DatabaseAccesser.js
Logging out now completely deletes the currentUser value in the 
localStorage
Removed unnecissary user object
Server now prints just the document's id

ZOrangeBandit reviewed (approved):
Seems fine to me

turtle11235 committed:
removed unnecessary files
